From 65c61af909e50da24ee1b07e222b6e4b9e37882d Mon Sep 17 00:00:00 2001
From: Shammah Chancellor <shammah.chancellor@gmail.com>
Date: Sun, 16 May 2021 19:21:06 -0700
Subject: [PATCH 103/196] Fix several mistakes in gitian/artifact bundling due
 to binary rename (#177)

In the zeal to rename everything to lotus, some changes to the gitian
linux and windows scripts were erroneously included. This commit reverts
those changes. Additionally, a few icon names were not correctly renamed
in the scripts despite the files being renamed. Longer term, more
references to bitcoin should be swapped to lotus to avoid confusion, but
within these scripts the references are immaterial.
---
 PackageOptions.cmake                        |  2 +-
 cmake/modules/NSIS.template.in              |  8 +--
 contrib/debian/lotus-qt.desktop             |  2 +-
 contrib/gitian-descriptors/gitian-linux.yml |  6 +--
 contrib/gitian-descriptors/gitian-osx.yml   |  2 +-
 contrib/gitian-descriptors/gitian-win.yml   | 56 ++++++++++-----------
 6 files changed, 38 insertions(+), 38 deletions(-)

diff --git a/PackageOptions.cmake b/PackageOptions.cmake
index 2b8a67c4b..ec6058667 100644
--- a/PackageOptions.cmake
+++ b/PackageOptions.cmake
@@ -37,7 +37,7 @@ set(_nsis_bitcoin_qt "lotus-qt.exe")
 set(CPACK_NSIS_URL_INFO_ABOUT "${CPACK_PACKAGE_HOMEPAGE_URL}")
 set(CPACK_NSIS_CONTACT "${CPACK_PACKAGE_CONTACT}")
 
-set(CPACK_NSIS_MUI_ICON "${CMAKE_SOURCE_DIR}/share/pixmaps/bitcoin-abc.ico")
+set(CPACK_NSIS_MUI_ICON "${CMAKE_SOURCE_DIR}/share/pixmaps/lotus.ico")
 set(CPACK_NSIS_MUI_UNIICON "${CPACK_NSIS_MUI_ICON}")
 set(CPACK_NSIS_MUI_WELCOMEFINISHPAGE_BITMAP "${CMAKE_SOURCE_DIR}/share/pixmaps/nsis-wizard.bmp")
 set(CPACK_NSIS_MUI_UNWELCOMEFINISHPAGE_BITMAP "${CPACK_NSIS_MUI_WELCOMEFINISHPAGE_BITMAP}")
diff --git a/cmake/modules/NSIS.template.in b/cmake/modules/NSIS.template.in
index 776fcde85..55b5adc4e 100644
--- a/cmake/modules/NSIS.template.in
+++ b/cmake/modules/NSIS.template.in
@@ -52,9 +52,9 @@ Var StartMenuGroup
 # Installer attributes
 OutFile "@CPACK_TOPLEVEL_DIRECTORY@/@CPACK_OUTPUT_FILE_NAME@"
 !if "@CPACK_SYSTEM_NAME@" == "x86_64-w64-mingw32"
-InstallDir $PROGRAMFILES64\Bitcoin-abc
+InstallDir $PROGRAMFILES64\Lotus
 !else
-InstallDir $PROGRAMFILES\Bitcoin-abc
+InstallDir $PROGRAMFILES\Lotus
 !endif
 CRCCheck on
 XPStyle on
@@ -107,7 +107,7 @@ Section -post SEC0001
     WriteRegDWORD HKCU "SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\$(^Name)" NoModify 1
     WriteRegDWORD HKCU "SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\$(^Name)" NoRepair 1
     WriteRegStr HKCR "@CPACK_PACKAGE_NAME@" "URL Protocol" ""
-    WriteRegStr HKCR "@CPACK_PACKAGE_NAME@" "" "URL:Bitcoin-abc"
+    WriteRegStr HKCR "@CPACK_PACKAGE_NAME@" "" "URL:Lotus"
     WriteRegStr HKCR "@CPACK_PACKAGE_NAME@\DefaultIcon" "" "$INSTDIR\${BITCOIN_GUI_NAME}"
     WriteRegStr HKCR "@CPACK_PACKAGE_NAME@\shell\open\command" "" '"$INSTDIR\${BITCOIN_GUI_NAME}" "%1"'
 SectionEnd
@@ -140,7 +140,7 @@ Section -un.post UNSEC0001
     Delete /REBOOTOK "$SMPROGRAMS\$StartMenuGroup\Uninstall $(^Name).lnk"
     Delete /REBOOTOK "$SMPROGRAMS\$StartMenuGroup\$(^Name).lnk"
     Delete /REBOOTOK "$SMPROGRAMS\$StartMenuGroup\@CPACK_NSIS_DISPLAY_NAME@ (testnet).lnk"
-    Delete /REBOOTOK "$SMSTARTUP\Bitcoin-abc.lnk"
+    Delete /REBOOTOK "$SMSTARTUP\Lotus.lnk"
     Delete /REBOOTOK "$INSTDIR\uninstall.exe"
     Delete /REBOOTOK "$INSTDIR\debug.log"
     Delete /REBOOTOK "$INSTDIR\db.log"
diff --git a/contrib/debian/lotus-qt.desktop b/contrib/debian/lotus-qt.desktop
index 27c61825d..11e3344f3 100644
--- a/contrib/debian/lotus-qt.desktop
+++ b/contrib/debian/lotus-qt.desktop
@@ -8,7 +8,7 @@ Comment[tr]=Lotus, eşten eşe kriptografik sanal para birimi
 Exec=lotus-qt %u
 Terminal=false
 Type=Application
-Icon=bitcoin-abc128
+Icon=lotus128
 MimeType=x-scheme-handler/payto;
 Categories=Office;Finance;
 StartupWMClass=Lotus-QT
diff --git a/contrib/gitian-descriptors/gitian-linux.yml b/contrib/gitian-descriptors/gitian-linux.yml
index 4041111c8..32dad3d74 100644
--- a/contrib/gitian-descriptors/gitian-linux.yml
+++ b/contrib/gitian-descriptors/gitian-linux.yml
@@ -1,5 +1,5 @@
 ---
-name: "lotusd-linux"
+name: "lotus-linux"
 enable_cache: true
 distro: "debian"
 suites:
@@ -34,7 +34,7 @@ repositories:
   - "cmake"
 remotes:
 - "url": "https://github.com/LogosFoundation/lotusd.git"
-  "dir": "lotus"
+  "dir": "bitcoin"
 files: []
 script: |
 
@@ -103,7 +103,7 @@ script: |
   create_per-host_faketime_wrappers "2000-01-01 12:00:00"
   export PATH=${WRAP_DIR}:${PATH}
 
-  cd lotus
+  cd bitcoin
   SOURCEDIR=`pwd`
   BASEPREFIX=`pwd`/depends
   # Build dependencies for each host
diff --git a/contrib/gitian-descriptors/gitian-osx.yml b/contrib/gitian-descriptors/gitian-osx.yml
index dedb443a8..7c1e65f62 100644
--- a/contrib/gitian-descriptors/gitian-osx.yml
+++ b/contrib/gitian-descriptors/gitian-osx.yml
@@ -1,5 +1,5 @@
 ---
-name: "bitcoin-abc-osx"
+name: "lotus-osx"
 enable_cache: true
 distro: "debian"
 suites:
diff --git a/contrib/gitian-descriptors/gitian-win.yml b/contrib/gitian-descriptors/gitian-win.yml
index ea7483d54..3c5f74877 100644
--- a/contrib/gitian-descriptors/gitian-win.yml
+++ b/contrib/gitian-descriptors/gitian-win.yml
@@ -3,34 +3,34 @@ name: "lotus-win"
 enable_cache: true
 distro: "debian"
 suites:
-- "buster"
+  - "buster"
 architectures:
-- "amd64"
+  - "amd64"
 packages:
-- "autoconf"
-- "automake"
-- "bsdmainutils"
-- "ca-certificates"
-- "curl"
-- "faketime"
-- "g++"
-- "g++-mingw-w64"
-- "git"
-- "libtool"
-- "ninja-build"
-- "mingw-w64"
-- "nsis"
-- "pkg-config"
-- "python3"
-- "zip"
+  - "autoconf"
+  - "automake"
+  - "bsdmainutils"
+  - "ca-certificates"
+  - "curl"
+  - "faketime"
+  - "g++"
+  - "g++-mingw-w64"
+  - "git"
+  - "libtool"
+  - "ninja-build"
+  - "mingw-w64"
+  - "nsis"
+  - "pkg-config"
+  - "python3"
+  - "zip"
 repositories:
-- "distribution": "buster-backports"
-  "source": "deb http://deb.debian.org/debian/ buster-backports main"
-  packages:
-  - "cmake"
+  - "distribution": "buster-backports"
+    "source": "deb http://deb.debian.org/debian/ buster-backports main"
+    packages:
+      - "cmake"
 remotes:
-- "url": "https://github.com/LogosFoundation/lotusd.git"
-  "dir": "lotus"
+  - "url": "https://github.com/LogosFoundation/lotusd.git"
+    "dir": "bitcoin"
 files: []
 script: |
   WRAP_DIR=$HOME/wrapped
@@ -84,7 +84,7 @@ script: |
 
   function create_per-host_linker_wrapper {
   # This is only needed for trusty, as the mingw linker leaks a few bytes of
-  # heap, causing non-determinism. See discussion in https://github.com/lotus/lotus/pull/6900
+  # heap, causing non-determinism. See discussion in //github.com/bitcoin/bitcoin/pull/6900
   for i in ${HOSTS[@]}; do
     mkdir -p ${WRAP_DIR}/${i}
     for prog in collect2; do
@@ -113,7 +113,7 @@ script: |
   create_per-host_linker_wrapper "2000-01-01 12:00:00"
   export PATH=${WRAP_DIR}:${PATH}
 
-  cd lotus
+  cd bitcoin
   SOURCEDIR=`pwd`
   BASEPREFIX=`pwd`/depends
   # Build dependencies for each host
@@ -169,7 +169,7 @@ script: |
       -DCLIENT_VERSION_IS_RELEASE=ON \
       -DENABLE_CLANG_TIDY=OFF \
       -DENABLE_REDUCE_EXPORTS=ON \
-      -DBUILD_lotus_SEEDER=OFF \
+      -DBUILD_BITCOIN_SEEDER=OFF \
       -DCPACK_STRIP_FILES=ON \
       -DCMAKE_INSTALL_PREFIX=${INSTALLPATH} \
       -DCCACHE=OFF \
@@ -182,7 +182,7 @@ script: |
 
     ninja package
 
-    cp -f lotusd-*-setup-unsigned.exe ${OUTDIR}/
+    cp -f ${DISTNAME}-*-setup-unsigned.exe ${OUTDIR}/
     cd installed
     mkdir -p ${DISTNAME}/lib
     mv ${DISTNAME}/bin/*.dll* ${DISTNAME}/lib/
-- 
2.39.5 (Apple Git-154)

